xtext/org.icyphy.linguafranca/src/org/icyphy/generator/GeneratorBase.xtend

```diff
@@ -1062,13 +1071,38 @@ abstract class GeneratorBase extends AbstractLinguaFrancaValidator {
         val bashReturn = bashBuilder.runSubprocess(#[bashOut], #[])
         if (bashReturn == 0) {
             println("SUCCESS")
-            // extract the full command from the output of which
-            val newCmd = bashOut.toString().trim()
+            return ExecutionEnvironment.BASH
+        }
+        reportError("The command " + cmd + " could not be found.\n" +
+                    "Make sure that your PATH variable includes the directory where " + cmd + " is installed.\n" +
+                    "You can set PATH in ~/.bash_profile on Linux or Mac.")
+        return null as ExecutionEnvironment
+            
+    }
+    
+    /**
+     * Creates a ProcessBuilder for a given command and its arguments.
+     * 
+     * This method returns correctly constructed ProcessBuilder object 
+     *according to the Execution environment. Raise an error if the env is null.
+     * 
+     * @param cmd The command to be executed
+     * @param args A list of arguments for the given command
+     * @param env is the type of the Execution Environment.
+     * @return A ProcessBuilder object if the command was found or null otherwise.
+     */
+    protected def createCommand(String cmd, List<String> args, ExecutionEnvironment env) { 
+        if (env == ExecutionEnvironment.NATIVE) {
+            val builder = new ProcessBuilder(#[cmd] + args)
+            builder.directory(new File(directory))
+            return builder
+        } else if (env == ExecutionEnvironment.BASH) {
+            val newCmd = #["bash", "--login", "-c", "\"", cmd]
```

Comment:
What is the `"\""` doing? It looks suspicious.

Reply:
In order for the command to work in windows we need to add quotation marks to gcc command. That's why I added \" as opening quotations.